From bcb4181eb0426492db4b6634ffd96b257117a2f2 Mon Sep 17 00:00:00 2001
From: Uldiniad <olivercscott@gmail.com>
Date: Thu, 10 May 2018 12:34:15 -0400
Subject: [PATCH 5/7] dex2oat: disable multithreading for WSL

* In its current state, WSL does not support dex2oat multithreading

Change-Id: I325d7f3428d74cb7a4845ef9cf339115a36ce832
---
 core/dex_preopt_libart.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/dex_preopt_libart.mk b/core/dex_preopt_libart.mk
index 15cb221b6..a81185baa 100644
--- a/core/dex_preopt_libart.mk
+++ b/core/dex_preopt_libart.mk
@@ -147,10 +147,14 @@ endif
 
 # $(1): the input .jar or .apk file
 # $(2): the output .odex file
+ifeq ($(HOST_OS_IS_WSL),true)
+SINGLE_THREAD := "-j1"
+endif
 define dex2oat-one-file
 $(hide) rm -f $(2)
 $(hide) mkdir -p $(dir $(2))
 $(hide) ANDROID_LOG_TAGS="*:e" $(DEX2OAT) \
+	$(SINGLE_THREAD) \
 	--runtime-arg -Xms$(DEX2OAT_XMS) --runtime-arg -Xmx$(DEX2OAT_XMX) \
 	--class-loader-context=$(DEX2OAT_CLASS_LOADER_CONTEXT) \
 	--boot-image=$(PRIVATE_DEX_PREOPT_IMAGE_LOCATION) \
-- 
2.17.1

